torch/csrc/api/include/torch/nn/modules/any.h

```diff
@@ -552,5 +565,80 @@ ModuleType& AnyModule::get_(
       c10::demangle(typeid(ModuleType).name()));
 }
 
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ NamedAnyModule ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+class NamedAnyModule {
```

Comment:
I think you should create a separate file for this. 600 lines is pretty long for a file and not everyone who needs `AnyModule` will need `NamedAnyModule`.